sqlglot/parser.py

```diff
@@ -1096,6 +1099,16 @@ def _parse_join(self):
 
         return self.expression(exp.Join, **kwargs)
 
+    def _parse_index(self):
+        index = self._parse_id_var()
+        self._match(TokenType.ON)
+        self._match(TokenType.TABLE)  # hive
+        # table = self._parse_table(schema=True)
```

Comment:
if you jus do parse_table(schema=True), then you don't even need columns because now the definition resides in the table schema.

although it's a bit more complex for the index use case, it makes sqlglot more consistent, is it ok if we stick with parse_table(schema=True) ?

Reply:
Well, "ok" is relative, I'd say. For me it would be okay, but if we use ``self._parse_table(schema=True)``, then suddenly this is "valid":

> CREATE UNIQUE INDEX IF NOT EXISTS my_idx ON TABLE tbl (a INT, b)

because it's a valid column definition, but it's definitely wrong SQL.

It has also an influence on the pretty printing.

Compare 

    CREATE UNIQUE INDEX IF NOT EXISTS my_idx ON tbl (a, b)

to

    CREATE UNIQUE INDEX IF NOT EXISTS my_idx ON tbl (
      a,
      b
    )

though that I don't mind, we're using a custom pretty printer anyway.